### ISSUE TITLE
Implement theming in component library

### ISSUE THREAD

**Issue by bradfordjanice** [hackoregon/civic#433]
Currently themes are kept with projects.  Using global themes would allow them to both be used in the component library and to style a project(s) by using it as a prop to that project.

An example of a theme:
civic/packages/component-library/src/VictoryTheme/CivicVictoryTheme.js

**Comment by jaronheard** [comment]
## How this might work

Emotion has theming, and I think implementing their theming is very possible:
https://emotion.sh/docs/emotion-theming

Here's how I imagine this working:

There would be a `civicTheme.js` file that is a nested object with all of the emotion styling information for the theme.

It would specify styling based on CIVIC's design system. It should be generic enough that other themes could be developed, and that if our component implementation changes that we can still use the information in this theme file.

This would incorporate the _component specific_ common styling that currently lives in `global.styles.css` and in `CivicVictoryTheme.js`. Each component would have an optional theme property, with a default theme, and be wrapped in a `<ThemeProvider>`.

Here's psuedocode for how it might be implemented in the 2018 package:
```js
import { civicTheme } from "@hackoregon/component-library";
...
<ThemeProvider theme={civicTheme}>
  <Global styles={theme.global}>
    <App />
  </Global>
</ThemeProvider>
```

For components, all components in the component library would take an optional theme prop, and be wrapped in ThemeProvider, something like this psuedocode:
```js
import civicTheme from "../civicTheme.js";
...
const PieChart = ({ theme }) =>
<ThemeProvide theme={theme}>
    <VictoryPie theme={theme.victoryTheme} />
</ThemeProvider>

PieChart.defaultProps = {
  theme: civicTheme,
}
```

**Comment by jaronheard** [comment]
Closed via #790
